### DISCUSSION TITLE
Where do the highlights for C come from?

### DISCUSSION THREAD

**Discussion by tristan957** [olimorris/onedarkpro.nvim#226] [Q&A] [answered]
They don't seem to match the VSCode theme. For instance, macros seem to have the same color as functions for me. Wanted to know if this was intended behavior or not. Otherwise I will customize them on my side.

**Comment by olimorris** [comment]
Must be coming from Tree-sitter as there are no custom C highlights in the plugin. If you share some code I'll be happy to take a look at the highlighting

**Comment by tristan957** [comment]
Neovim:

![image](https://github.com/olimorris/onedarkpro.nvim/assets/11936772/b19f5c34-d7c9-4568-b1cc-4610b9ea4545)

VSCode:

![image](https://github.com/olimorris/onedarkpro.nvim/assets/11936772/250d50b8-297a-431e-8209-acbb0353f79a)

I am a heavy C user FWIW, so I can definitely help match C up with VSCode if I know where to look!

**Reply by tristan957** [reply]
```c
#include <stddef.h>
#include <stdio.h>

#define HELLO_WORLD "hello world"

int
main(int argc, char **argv)
{
    printf("%s: %p", HELLO_WORLD, NULL);
    if (NULL) {
        return 0;
    } else {
        return 1;
    }
}
```

**Reply by olimorris** [reply]
Check out the [CONTRIBUTING.md](https://github.com/olimorris/onedarkpro.nvim/blob/main/CONTRIBUTING.md) file. Would welcome a PR for this!

I typically ask ChatGPT to generate an example file containing common syntax examples and then paste that into VS Code and Neovim to use as a template to work from.

**Reply by tristan957** [reply]
Looks very easy to get started. Hopefully I will come up with a PR soon!

**Comment by tristan957** [comment]
https://github.com/olimorris/onedarkpro.nvim/pull/227

**Reply by dbozbay** [reply]
Would it be possible to extend the custom C highlights to cpp files?

**Comment by tristan957** [comment]
@olimorris I was wondering what the stance on the project is when a theme diverges from VSCode. Is it a bug if they don't match? I was just looking at a Markdown file which is colored quite differently from VSCode.

**Reply by olimorris** [reply, marked as answer]
Really good question. Generally speaking, I try and stay as close to the VS Code theme as possible but Markdown is currently an exception. I can't recall when it diverged so much but it's quite a bit different now.

It feels a bit strong to call the divergence a bug but technically it is so would be open to any pull requests (and for Markdown I will fix that myself). The theme makes it easy for users to apply their own custom highlights anyway.

**Reply by tristan957** [reply]
Very true. Just thought I would upstream any changes if it is the goal to match.

**Reply by olimorris** [reply]
Yeah please do! If you could provide before and after screenshots in any PRs that would be awesome

**Reply by tristan957** [reply]
Cool. Are you still going to do the Markdown updates? I have noticed C++ files vary too. Python imports are also uncolored. I can try to get to both of those some time soon.

**Reply by olimorris** [reply]
Yeah I'll take the markdown ones as I'll provide a guide for users to revert to the previous colors.

But feel free to get started on any of the others. Note...they may require some custom Tree-sitter queries
